### PR TITLE
chore: Small fixes on for deb and rpm targets

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Caught smol things while working on the target spec implementations, kicked them out to a separate PR as they're unrelated

### CODE REVIEW

File: distribution/rpm/vector.spec [hunk 1]
```diff
@@ -66,6 +66,7 @@ getent passwd %{_username} > /dev/null || \
     --comment "Vector observability data router" %{_username}
 chown %{_username} %{_sharedstatedir}/%{_name}
 usermod -aG systemd-journal %{_username}  || true
+usermod -aG systemd-journal-remote %{_username}  || true
```
Comment (spencergilbert) [inline distribution/rpm/vector.spec:69]: This was added by a contributor to the deb package, mirrored here for completeness.

File: distribution/rpm/vector.spec [hunk 2]
```diff
@@ -31,7 +31,7 @@ Name: %{_name}
 Summary: A lightweight and ultra-fast tool for building observability pipelines
 Version: %{_cleaned_version}
 Release: %{_release}
-License: ASL 2.0
+License: MPL-2.0
```
Comment (jszwedko) [inline distribution/rpm/vector.spec:34]: 🤦

**Review by jszwedko — APPROVED** [review]
Thanks for the cleanups!

File: distribution/rpm/vector.spec (hunk 2 above)
Comment (spencergilbert) [reply distribution/rpm/vector.spec:34]: (We didn't update our nix one either, but I just deleted that one). I need to PR or open an issue for the community nix package, which also lists us as ASL2.0

**Pull request merged**